cmake/external/mkldnn.cmake

```diff
@@ -35,11 +35,28 @@ SET(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_RPATH}" "${MKLDNN_INSTALL_DIR}/${LIBDIR
 
 INCLUDE_DIRECTORIES(${MKLDNN_INC_DIR}) # For MKLDNN code to include internal headers.
 
+IF(${CBLAS_PROVIDER} STREQUAL "MKLML")
+    SET(MKLDNN_DEPENDS   ${MKLML_PROJECT})
+    MESSAGE(STATUS "Build MKLDNN with MKLML ${MKLML_ROOT}")
+ELSE()
+    MESSAGE(STATUS "Build MKLDNN without MKLML")
+ENDIF()
+
 IF(NOT WIN32)
     SET(MKLDNN_FLAG "-Wno-error=strict-overflow -Wno-error=unused-result -Wno-error=array-bounds")
     SET(MKLDNN_FLAG "${MKLDNN_FLAG} -Wno-unused-result -Wno-unused-value")
     SET(MKLDNN_CFLAG "${CMAKE_C_FLAGS} ${MKLDNN_FLAG}")
     SET(MKLDNN_CXXFLAG "${CMAKE_CXX_FLAGS} ${MKLDNN_FLAG}")
+
+    IF(${CBLAS_PROVIDER} STREQUAL "MKLML")
+    # Force libmkldnn.so to link libiomp5.so (provided by intel mkl) instead of libgomp.so (provided by gcc),
+    # since core_avx.so links libiomp5.so 
+        set(MKLDNN_SHARED_LINKER_FLAG "${CMAKE_SHARED_LINKER_FLAGS} -Wl,--no-as-needed -L${MKLML_LIB_DIR} -liomp5")
+        set(FORBID "-fopenmp")
```

Comment:
In (https://github.com/intel/mkl-dnn/issues/625) it was said that setting `-fopenmp` flag is not reliable solution and its outcome depends on CMake version.

Reply:
Yes, I think this can solve part of problems as it is used in [OpenMP.cmake] (https://github.com/intel/mkl-dnn/blob/master/cmake/OpenMP.cmake).
And we need to find a further reliable solution if we can.

Reply:
When this compilation issue happen? Is there some specific configuration that fails or it's more common?

Reply:
After PR #20162, core_avx.so links libiomp5.so and  libgomp5.so.
Before that, it only links libiomp5.so.